### DISCUSSION TITLE
List of external completion scripts?

### DISCUSSION THREAD

**Discussion by akinomyoga** [scop/bash-completion#550] [General]
Maybe we can manage a list of external completion scripts with the link to their upstream. The list doesn't need to be complete, but we add an item when someone notices a new external completion script.

It is useful when we want to notify the external completion writers of the change in `bash-completion`. For example, in https://github.com/scop/bash-completion/pull/531#issuecomment-853557538, the `slackpkg` maintainer wanted notifications in the future bash-completion changes that may break the `slackpkg` completion. We don't intend to break the existing completion scripts by changing bash-completion behavior, but it is still useful to notify the external completion writers about deprecations of some interfaces and introduction of newer and more useful interfaces.

**Comment by scop** [comment]
Sounds good to me, started https://github.com/scop/bash-completion/wiki/Externally-available-completions for this. Lots of stuff to backfill, and feel free to adjust the formatting etc as you see fit.

**Reply by scop** [reply]
Not sure what's the best way to mention projects with lots of completions, but I added a oneliner for util-linux there now. For correctness I guess a row per executable would be best, but dunno if that's the best wrt keeping things up to date.

The intent for sorting by executable version is to help people locate commands they're interested in, but listing several in the same column kind of defeats the purpose and requires Ctrl+F or the like anyway. So that would kind of point towards entry per executable, too.

**Comment by scop** [comment]
I see you added a few more tables and more info there, cool. Just one question/clarification about the "independent of bash-completion" list: is the intent of that to be such that the completions listed there do not make use of anything in bash-completion? Asking because if that's the case, we may want to make it more clear that things in the first list _do_ have dependencies, and because util-linux in the first list seems like it would belong to the "independent" list.

Maybe formattingwise it could be better to combine those in the same list and just add a checkbox column whether something is known to have dependencies to bash-completion or not. And also include that info for generators.

**Reply by akinomyoga** [reply]
> I see you added a few more tables and more info there, cool. Just one question/clarification about the "independent of bash-completion" list: is the intent of that to be such that the completions listed there do not make use of anything in bash-completion?

Ah, sorry for the unclear description. Yes, they are the completions that don't have dependencies on `bash-completion`.

> Asking because if that's the case, we may want to make it more clear that things in the first list do have dependencies, and because util-linux in the first list seems like it would belong to the "independent" list.

I see. OK.

> Maybe formattingwise it could be better to combine those in the same list and just add a checkbox column whether something is known to have dependencies to bash-completion or not. And also include that info for generators.

Ah, right. The generated completions also have these two categories with/without dependencies. It is better to combine them rather than having 2x2=4 tables. I have combined the tables. Also, since the width of the wiki page is limited, I collapsed some columns.

By the way, it seems `[ ] ` or `- [ ] ` in a table cell won't be rendered as checkboxes, and there are [various alternative ways](https://stackoverflow.com/questions/47344571/how-to-draw-checkbox-or-tick-mark-in-github-markdown-table/55523035). I tried several ways.

| Checkbox type | unchecked | checked |
|:--|:--:|:--:|
| `- [ ] ` | - [ ] a | - [x] b |
| `<ul><li>[ ]` | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> |
| `<input type="checkbox" />` | <input type="checkbox" disabled /> | <input type="checkbox" disabled checked /> |
| `&#9745;` | &#9744;   | &#9745; |
| `&#x2705;` | &#x2B1C; | &#x2705; |
| `:heavy_check_mark:` | :white_check_mark: | :heavy_check_mark: |
| `:white_check_mark:` | :white_large_square: | :white_check_mark: |
| ✅ | ⬜️ | ✅ |

The rendering result seems to even differ in GitHub Issue comments and in GitHub Wiki pages.

**P.S.** I think the result also depends on the Web browser and the operating system.

I finally decided to use `&#x2B1C;` and `&#x2705;`.

**Reply by scop** [reply]
Great stuff, thanks!
